### PR TITLE
ES-2476: Fix JSON format logging in Corda CLI

### PR DESCRIPTION
**Pull request by anton-subbotin**
### [ES-2476](https://r3-cev.atlassian.net/browse/ES-2476)

JSON format logging in Corda CLI got broken after https://github.com/corda/corda-runtime-os/pull/6169

This PR brings in the missing pieces of Gradle configuration.

[ES-2476]: https://r3-cev.atlassian.net/browse/ES-2476?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: tools/corda-cli/build.gradle [hunk 1]
```diff
@@ -142,8 +142,30 @@ if (project.hasProperty(S3_BUCKET_URI_PROPERTY)) {
...
+
+    doLast {
+        def inputFiles = inputs.files.getFiles().findAll { it.name == 'Log4j2Plugins.dat' }.collect { it.toURI().toURL() }
+        if (inputFiles) {
+            def combinedCache = new org.apache.logging.log4j.core.config.plugins.processor.PluginCache()
+            combinedCache.loadCacheFiles(Collections.enumeration(inputFiles))
+            def outputFile = outputDir.get().file('META-INF/org/apache/logging/log4j/core/config/plugins/Log4j2Plugins.dat').asFile
+            outputFile.parentFile.mkdirs()
+            try (OutputStream out = new FileOutputStream(outputFile)) {
+                combinedCache.writeCache(out)
```
Comment (anton-subbotin) [inline tools/corda-cli/build.gradle:159]: Although we have the same piece of logic as a part of `corda.common-app` plugin in `buildSrc`, reusing it here would be a massive gradle configuration rework of corda-cli app to be built using OSGi framework, with the overhead of all other stuff coming from the plugin which is required for the workers but unnecessary in Corda CLI.

File: tools/corda-cli/build.gradle [hunk 2]
```diff
@@ -142,8 +142,30 @@ if (project.hasProperty(S3_BUCKET_URI_PROPERTY)) {
     }
 }
 
+def mergeLog4j2Plugins = tasks.register('mergeLog4j2Plugins') {
+    dependsOn configurations.runtimeClasspath
+    inputs.files(configurations.runtimeClasspath.collect { it.isDirectory() ? it : zipTree(it) })
```
Comment (anton-subbotin) [inline tools/corda-cli/build.gradle:150]: Although we have the same piece of logic as a part of `corda.common-app` plugin in `buildSrc`, reusing it here would be a massive gradle configuration rework of corda-cli app to be built using OSGi framework, with the overhead of all other stuff coming from the plugin which is required for the workers but unnecessary in Corda CLI.

**Comment by anton-subbotin** [comment]
E2E tests: https://ci02.dev.r3.com/blue/organizations/jenkins/Corda5%2Fcorda5-e2e-tests/detail/PR-680/2/pipeline

File: tools/corda-cli/build.gradle (hunk 2 above)
Comment (tlawson3) [reply tools/corda-cli/build.gradle:150]: if this is the case, fair enough but it looks to me like the task in corda.common-app.gradle is a top-level task, not nested inside any of the classes doing the OSGi stuff.
I had hoped that you could add `id 'corda.common-app'` and then rely on this task.

Is there a way to move the task mergeLog4j2Plugins into a new separate buildSrc file, which both this and common-app can share?

Comment (anton-subbotin) [reply tools/corda-cli/build.gradle:150]: > add id 'corda.common-app' and then rely on this task

Did that, not only I'm getting a bunch of errors unable to export many various Corda classes which the Corda CLI has no direct use of, but the `jar` task configuration of Corda CLI conflicts with the `appJar` task provided by the `corda.common-app` plugin as its main functionality.
Was also trying to reuse `appJar` instead, but that means reworking the main App class and not use `id 'application'` plugin, but implement the OSGi Application etc etc.

> Is there a way to move the task mergeLog4j2Plugins into a new separate buildSrc file, which both this and common-app can share?

Trying that now, not much success so far..

Comment (vkolomeyko) [reply tools/corda-cli/build.gradle:150]: Perhaps, we should add a comment as for why `mergeLog4j2Plugins` task is necessary. E.g. why simply including JSON Log4J layout into the class path is not enough.

Comment (anton-subbotin) [reply tools/corda-cli/build.gradle:150]: Spent some more time trying to reuse the task logic from the `buildSrc`, no luck. Adding the comments explaining the problem.

File: tools/corda-cli/build.gradle [hunk 3]
```diff
@@ -210,10 +236,3 @@ publishing {
         }
     }
 }
-
-tasks.named("installDist") {
```
Comment (tlawson3) [inline tools/corda-cli/build.gradle:214]: Why is this being removed?

Comment (anton-subbotin) [reply tools/corda-cli/build.gradle:214]: Not used. Was added as a part of migration, but actually has no useful effect and isn't being used anywhere. 

Originally each of the plugins included such a task which would copy the plugin's jar into the installation path's plugins dir `~/.corda/cli/plugins`. With the new architecture there's no sense to do that, as a different and correct installation procedure is available for Corda CLI as a whole.

**Review by tlawson3 — APPROVED** [review]

**Review by vkolomeyko — APPROVED** [review]
LGTM

**Pull request merged**